src/common/ant.ts

```diff
@@ -87,6 +91,99 @@ export class ANT {
    */
   static fork = forkANT;
 
+  /**
+   * Upgrade an ANT by forking it to the latest version and reassigning names.
+   *
+   *
+   * @param config Configuration object for the upgrade process
+   * @returns Promise resolving to the forked process ID and successfully reassigned names
+   */
+  static async upgrade({
+    signer,
+    antProcessId,
+    names,
+    arioProcessId = ARIO_MAINNET_PROCESS_ID,
+    ao,
+    logger = Logger.default,
+    antRegistryId,
+    skipVersionCheck = false,
+    antProcess = ANT.init({
+      process: new AOProcess({ processId: antProcessId, ao }),
+      signer,
+    }),
+    onSigningProgress,
```

Comment:
I would recommend we:
- make this a non static method, leveraging the set signer, process id, aoClient, and logger, etc.
- add module ID and tags as options - tags good to track what integration is upgrading, module id for CLI usage and testing / custom versions.
- hyperbeam URL for leveraging cached state

Reply:
there is already a non-static method that calls this method